pkgs/development/haskell-modules/configuration-ghc-9.8.x.nix

```diff
@@ -48,27 +49,119 @@ self: super: {
   unix = null;
   xhtml = null;
 
-  # https://github.com/tibbe/unordered-containers/issues/214
-  unordered-containers = dontCheck super.unordered-containers;
+  aeson = super.aeson_2_2_1_0;
+  aeson-pretty = super.aeson-pretty_0_8_10;
+  attoparsec-aeson = super.attoparsec-aeson_2_2_0_1;
+  alex = super.alex_3_4_0_1;
+  bifunctors = super.bifunctors_5_6_1;
+  cabal-install-solver = super.cabal-install-solver_3_10_2_1;
+  doctest = super.doctest_0_22_2;
+  free = super.free_5_2;
+  # ghc-lib 9.8.1.20231121 required for Cabal to build: https://github.com/digital-asset/ghc-lib/issues/495
+  ghc-lib = super.ghc-lib_9_8_1_20231121;
+  ghc-lib-parser = super.ghc-lib-parser_9_8_1_20231121;
+  ghc-lib-parser-ex = super.ghc-lib-parser-ex_9_8_0_0;
+  github = super.github_0_29;
+  hedgehog = super.hedgehog_1_4;
+  hspec = super.hspec_2_11_7;
+  hspec-core = super.hspec-core_2_11_7;
+  hspec-discover = super.hspec-discover_2_11_7;
+  hspec-hedgehog = super.hspec-hedgehog_0_1_1_0;
+  hspec-meta = super.hspec-meta_2_11_7;
+  megaparsec = super.megaparsec_9_6_1;
+  megaparsec-tests = super.megaparsec-tests_9_6_1;
+  ormolu = super.ormolu_0_7_3_0;
+  rebase = super.rebase_1_20_1_1;
+  rerebase = super.rerebase_1_20_1_1;
+  semigroupoids = super.semigroupoids_6_0_0_1;
+  singleton-bool = super.singleton-bool_0_1_7;
+  some = super.some_1_0_6;
+  tagged = super.tagged_0_8_8;
+  tasty-hspec = super.tasty-hspec_1_2_0_4;
+  th-abstraction = super.th-abstraction_0_6_0_0;
+
+  cabal-fmt =
+    if pkgs.stdenv.targetPlatform.isDarwin && pkgs.stdenv.targetPlatform.isAarch64 then
+      overrideCabal (drv: {
+        enableSeparateBinOutput = false;
```

Comment:
Encountered https://github.com/NixOS/nixpkgs/issues/140774 but for `cabal-fmt` and setting `enableSeparateBinOutput` actually worked. Not sure if this is cause for concern?

Reply:
For a matching fix to the other GHC versions you'd need to port `pkgs/development/compilers/ghc/Cabal-3.6-paths-fix-cycle-aarch64-darwin.patch` to GHC 9.8. A similar PR for GHC 9.6 is in limbo at the moment: https://github.com/NixOS/nixpkgs/pull/258280.

Reply:
You can drop this override, since #258280 should fix this issue. I'd be grateful for any tests of that PR you could do, I currently don't have access to the appropriate hardware.

Reply:
Got merged, so if you rebase on haskell-updates, you can just drop this patch.